### PR TITLE
Improved profiler

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Adding few missing things to profiler like widget performance and better showing details for every module
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | n/a
| How to test?      | use profiler, compare current behavior to old one, dev qa only IMO
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.
| Sponsor company | impSolutions

Before:
https://www.dropbox.com/s/gpbfrb7zbstcb9e/Zrzut%20ekranu%202021-05-26%20o%2010.09.11.png?dl=0

After:
https://www.dropbox.com/s/ifkocyoj5y0jozq/Zrzut%20ekranu%202021-05-26%20o%2010.13.32.png?dl=0

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24659)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]
:heart:

**Comment by PierreRambaud** [comment]
FYI, I asked him to target the 1.7.8.x because a lot of work has been done in this version and these missing information are considered as regressions :)
Great job @kpodemski

File: tools/profiling/Profiler.php [hunk 1]
```diff
@@ -30,6 +30,8 @@ class Profiler
     protected $hooksPerfs = [];
     protected $modulesPerfs = [];
     protected $profiler = [];
+    protected $globalVarSize = [];
+    protected $queries = [];
```
Comment (Progi1984) [inline tools/profiling/Profiler.php]: Please add PHPDoc

File: tools/profiling/Profiler.php [hunk 2]
```diff
@@ -39,6 +41,7 @@ class Profiler
     protected $totalHooksTime = 0;
     protected $totalHooksMemory = 0;
     protected $startTime = 0;
+    protected $totalCacheSize = 0;
```
Comment (Progi1984) [inline tools/profiling/Profiler.php:69]: Please add PHPDoc

File: tools/profiling/Hook.php [hunk 3]
```diff
@@ -44,6 +44,44 @@ public static function coreCallHook($module, $method, $params)
...
+                'method' => $method,
+                'time' => microtime(true) - $timeStart,
+                'memory' => memory_get_usage() - $memoryStart,
+            ]
+        );
+
+        return $result;
+    }
+
+    public static function coreRenderWidget($module, $registeredHookName, $params)
```
Comment (Progi1984) [inline tools/profiling/Hook.php:59]: Could you typeint this new method ?

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: tools/profiling/Hook.php (hunk 3 above)
Comment (kpodemski) [reply tools/profiling/Hook.php:59]: I don't think I can as there is a warning about method not compatible with parent :)

Comment (Progi1984) [reply tools/profiling/Hook.php:59]: If it extends, you can. If it was an implementation (interface), you can't.

Comment (kpodemski) [reply tools/profiling/Hook.php:59]: @Progi1984 

https://www.dropbox.com/s/mtm385bi96rgl5z/Zrzut%20ekranu%202021-05-26%20o%2015.39.48.png?dl=0

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
@Robin-Fischer-PS I let you decide if the test has to be done by devs :)

**Comment by Robin-Fischer-PS** [comment]
Thanks for the ping @PierreRambaud ! I think it's best if this is tested by dev. I add the label.

:heart:

**Comment by PierreRambaud** [comment]
Great job @kpodemski Thanks again

**Pull request merged**